### PR TITLE
20 make slides dots and arrows forms for interactive example

### PR DESCRIPTION
**Pull request by IgorKozlovsky**

### CODE REVIEW

**Review by alex-dishen — APPROVED** [review]

File: src/components/NumberInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <Box>
+      <Text maxWidth="140px" color="gray.200" pb="2" fontSize="lg">
+        {title}
+      </Text>
+      <NumberWrapper>
+        <NumberArrows>
+          <ArrowUp
+            style={{ cursor: 'pointer' }}
+            onClick={() => handleClick('+')}
```
Comment (MaKs-Tkachyk) [inline src/components/NumberInput/index.tsx:29]: I think will be better rename from '+' to increase for more understanding
```
handleClick('increase')
handleClick('decrease')
```

File: src/components/NumberInput/styles.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { css } from '@emotion/react'
+import styled from '@emotion/styled'
+
+export const NumberWrapper = styled(Box)(
+  ({ theme }) => css`
+    display: flex;
+    position: relative;
+    align-items: center;
+    justify-content: center;
+    font-size: ${theme.fontSizes.lg};
```
Comment (MaKs-Tkachyk) [inline src/components/NumberInput/styles.ts:11]: font-size: ${theme.fontSizes.lg}; after  background-color: ${theme.colors.gray[900]};

File: src/components/NumberInput/styles.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    background-color: ${theme.colors.gray[900]};
+    border-radius: ${theme.radii.lg};
+  `
+)
+
+export const StyledNumberInput = styled('input')(
+  ({ theme }) => css`
+    width: 85px;
+    height: 100%;
+    padding: ${theme.space[1]};
```
Comment (MaKs-Tkachyk) [inline src/components/NumberInput/styles.ts:23]: padding: ${theme.space[1]}; before sizes

File: src/components/NumberInput/styles.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    gap: ${theme.space[2]};
+    padding-left: ${theme.space[2]};
+    left: 0px;
+  `
+)
+
+export const NumberUnit = styled('span')(
+  ({ theme }) => css`
+    font-size: ${theme.fontSizes.lg};
+    color: ${theme.colors.gray[500]};
```
Comment (MaKs-Tkachyk) [inline src/components/NumberInput/styles.ts:48]: color first

File: src/components/NumberInput/useInputNumber.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { ChangeEvent, useState } from 'react'
+
+export const useInputNumber = (defaultValue: number) => {
+  const [number, setNumber] = useState<number>(defaultValue)
+
+  const handleClick = (type: '+' | '-') => {
+    setNumber(type === '+' ? number + 1 : number - 1)
+  }
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```
Comment (MaKs-Tkachyk) [inline src/components/NumberInput/useInputNumber.ts:10]: return type

File: src/components/NumberInput/useInputNumber.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { ChangeEvent, useState } from 'react'
+
+export const useInputNumber = (defaultValue: number) => {
+  const [number, setNumber] = useState<number>(defaultValue)
+
+  const handleClick = (type: '+' | '-') => {
```
Comment (MaKs-Tkachyk) [inline src/components/NumberInput/useInputNumber.ts:6]: rewrite it how mentioned before

File: src/components/OptionsInput/styles.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    color: white;
+    font-size: ${theme.fontSizes.lg};
+    border-radius: ${theme.radii.lg};
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    appearance: none;
+    background-image: url('src/assets/arrow-select.svg');
+    background-repeat: no-repeat;
+    background-position-x: calc(100% - ${theme.space[2]});
+    background-position-y: center;
```
Comment (MaKs-Tkachyk) [inline src/components/OptionsInput/styles.ts:19]: order

File: src/components/OptionsInput/useInputOptions.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { ChangeEvent, useState } from 'react'
+
+export const useInputOptions = (defaultValue: string) => {
+  const [option, setOption] = useState<string>(defaultValue)
+
+  const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
```
Comment (MaKs-Tkachyk) [inline src/components/OptionsInput/useInputOptions.ts:6]: return type

File: src/components/Switcher/styles.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import { SwitcherButtonProps } from 'components/Switcher/types'
+
+export const SwitcherContainer = styled(Flex)(
+  ({ theme }) => css`
+    align-items: center;
+    width: fit-content;
+    padding: ${theme.space[1]};
+    gap: ${theme.space[1]};
+    background-color: ${theme.colors.gray[900]};
+    border-radius: ${theme.radii.lg};
```
Comment (MaKs-Tkachyk) [inline src/components/Switcher/styles.ts:13]: order

File: src/components/Switcher/styles.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  `
+)
+
+export const SwitcherButton = styled(Button)<SwitcherButtonProps>(
+  ({ theme, active, count }) => css`
+    min-width: ${count === 2 ? '185px' : '125px'};
+    font-size: ${theme.fontSizes.xl};
+    background-color: ${active ? theme.colors.gray[600] : 'transparent'};
+    color: ${active ? 'white' : theme.colors.gray[500]};
+    border-radius: ${theme.radii.lg};
```
Comment (MaKs-Tkachyk) [inline src/components/Switcher/styles.ts:23]: order

File: src/components/ColorInput/styles.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  `
+)
+
+export const StyledColorInput = styled.input(
+  ({ theme }) => css`
+    height: 45px;
+    padding: 0;
+    margin: 0;
+    background-color: ${theme.colors.gray[900]};
+    background-color: transparent;
```
Comment (IgorKozlovsky) [inline src/components/ColorInput/styles.ts:41]: 2 background-colors

File: src/components/OptionsInput/useInputOptions.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
+import { useState, useRef } from 'react'
+import { useClickOutside } from 'hooks/useClickOutside'
+
+export const useInputOptions = (defaultValue: string) => {
+  const [option, setOption] = useState<string>(defaultValue)
+  const [isOpenDropDown, setIsOpenDropDown] = useState<boolean>(false)
+  const dropDownRef = useRef<HTMLDivElement>(null)
```
Comment (IgorKozlovsky) [inline src/components/OptionsInput/useInputOptions.ts:7]: Let's set this order like this and add enter between different hooks:
useRef

useState
useState

File: src/components/NumberInput/styles.ts [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+  ({ theme }) => css`
+    position: absolute;
+    left: 0px;
+    flex-direction: column;
+    padding-left: ${theme.space[2]};
+    gap: ${theme.space[2]};
+  `
+)
+
+export const NumberUnit = styled('span')(
```
Comment (midstem) [inline src/components/NumberInput/styles.ts:45]: use Text

File: src/components/Switcher/index.tsx [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+const Switcher = ({
+  title,
+  buttons,
+  defaultValue = 0
+}: SwitcherProps): JSX.Element => {
+  const { activeValue, handleClick } = useSwitcher(buttons[defaultValue])
+
+  return (
+    <Box>
+      <Text color="gray.200" pb="2" fontSize="lg">
```
Comment (midstem) [inline src/components/Switcher/index.tsx:15]: good)

File: src/components/Switcher/styles.ts [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    background-color: ${theme.colors.gray[900]};
+    border-radius: ${theme.radii.lg};
+  `
+)
+
+export const SwitcherButton = styled(Button)<SwitcherButtonProps>(
+  ({ theme, active, count }) => css`
+    height: 37px;
+    width: ${count === 2 ? '144px' : '90px'};
+    background-color: ${active ? theme.colors.gray[600] : 'transparent'};
```
Comment (midstem) [inline src/components/Switcher/styles.ts:21]: rewrite  
```
 &, &:hover {
      background-color: ${active ? theme.colors.gray[600] : 'transparent'};
    }
```

File: src/features/Header/styles.ts [hunk 16]
```diff
@@ -36,11 +36,11 @@ export const StyledLogo = styled(Logo)(
 
 export const Text = styled.p(
```
Comment (midstem) [inline src/features/Header/styles.ts:37]: text and value from theme

File: src/features/Header/styles.ts [hunk 17]
```diff
@@ -55,10 +55,10 @@ export const Text = styled.p(
 export const NinjaWrapper = styled.div(
```
Comment (midstem) [inline src/features/Header/styles.ts:55]: box

File: src/features/Header/styles.ts [hunk 18]
```diff
@@ -79,13 +79,13 @@ export const StyledNinja = styled(Ninja)`
 
 export const BackLight = styled.div(
```
Comment (midstem) [inline src/features/Header/styles.ts:80]: box

File: src/features/SettingsTabs/ui/Dots.tsx [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+import Switcher from 'components/Switcher'
+import { dotsAnimationOptions } from '../constants'
+
+const Dots = (): JSX.Element => (
+  <StyledTabPanel>
+    <StyledTabPanelFlex>
+      <Switcher title="Show dots" buttons={['Yes', 'No']} />
+      <Switcher
+        title="Custom dot"
+        buttons={['Circle', 'Unicorn', 'None']}
```
Comment (midstem) [inline src/features/SettingsTabs/ui/Dots.tsx:18]: make enum for all button tiitles

**Pull request merged**